SIP-029: Preserving Economic Incentives During Stacks Network Upgrades (#196)
stacksgov/sips: sips/sip-029/sip-029-halving-alignment.md

# Preamble

**SIP Number:** 029

**Title:** Bootstrapping sBTC Liquidity and Nakamoto Signer Incentives

**Authors:**
- Alex Miller (alex@hiro.so)
- Andre Serrano (andre@bitcoinl2labs.com)
- Brittany Laughlin (brittany@lattice.vc)
- Jesse Wiley (jesse@stacks.org)
- Jude Nelson (jude@stacks.org)
- Philip De Smedt (philip@stackingdao.com)
- Tycho Onnasch (tycho@zestprotocol.com)
- Will Corcoran (will@stacks.org)

> **Consideration:** Economics, Technical, Governance

Reviewer: ```suggestion
**Consideration:** Economic, Technical, Governance
```
fix typo

> **Type:** Consensus (hard fork)

Reviewer: ```suggestion
**Type:** Consensus

**Layer:** Consensus (hard fork)
```

Reviewer 2: addressed

> 

Reviewer: Missing 'Layer' field in the Preamble: Since the SIP Type is 'Consensus (hard fork)', the Preamble must include the 'Layer' field set to 'Consensus (hard fork)' as per the SIP Specification.

Reviewer 2: updated

**Status:** Draft

**Created:** 2024-11-06

**License:** BSD 2-Clause

**Sign-off:**

**Discussions-To:**
- Stacks Forum Post: [Aligning with Bitcoin Halving and Incentives after Nakamoto](https://forum.stacks.org/t/aligning-with-bitcoin-halving-and-incentives-after-nakamoto/17668)

# Abstract

The first Stacks halving is expected to take place at Stacks block height 210,384, which is set to occur during Reward Cycle 100 in December 2024, cutting the STX block reward from 1,000 STX to 500 STX. This SIP proposes a modification to the emissions schedule given that the network is going through two major launches (Nakamoto and sBTC) which rely on predictable economic incentives for chain security. The proposed schedule modification and associated STX emission rate would create time for Nakamoto and sBTC to launch and settle in, but, being mindful of supply, would still result in an overall reduced target 2050 STX supply (0.19% lower) and a reduced tail emission rate (20% lower).

> 

Reviewer: missing *Copyrights* section

Reviewer 2: ```suggestion

# License and Copyright

This SIP is made available under the terms of the BSD-2-Clause license,
available at https://opensource.org/licenses/BSD-2-Clause.  This SIP’s copyright
is held by the Stacks Open Internet Foundation.

```

First, the role of Stackers is changing. The role of Stackers in the Nakamoto blockchain (SIP-021) differs from their role in the original Stacks blockchain in that they must now collectively sign Stacks blocks. This is required to prevent Stacks forks from arising, and to secure the chain tip before it can be fully anchored to the Bitcoin chain. This also means that Stackers must run and maintain signing nodes with high availability in order to ensure that blocks always reach the requisite signature threshold. The PoX payouts (SIP-007) granted to Stackers by miners provide a positive monetary incentive for Stackers to carry out this task.

Second, the release of the Nakamoto blockchain unblocks sBTC (SIP-028), a wrapped Bitcoin asset on the Stacks chain which benefits from Nakamoto’s newfound resistance to forks. Because inducing a chain reorganization (a reorg) in Nakamoto is at least as hard as doing so in the Bitcoin chain, applications can rely on Nakamoto to ensure that each sBTC token is backed 1:1 by a BTC token – there is no feasible way to double-spend sBTC through a Stacks chain reorg.

> Because of how important both of these upgrades are to the future of the Stacks blockchain and the dependency of all blockchains on economic incentives for security, changing the incentives while the system is going through transition could have negative impacts. In particular, given that miners bid BTC in order to win the STX coinbase reward (and that BTC is what provides PoX payouts for Signers), it is highly likely that a 50% reduction in the STX coinbase reward would lead to a corresponding 50% reduction PoX payouts, thereby dramatically decreasing incentives for signers while Nakamoto and sBTC are gaining adoption. Additionally, as discussed in the 7th Avenue Group report [1], by reducing the gross value of the block rewards and corresponding PoX payouts, it is likely that some miners and signers may choose to cease their work on the Stacks blockchain, reducing competition and economics further.

Reviewer: ```suggestion
Because of how important both of these upgrades are to the future of the Stacks blockchain and the dependency of all blockchains on economic incentives for security, changing the incentives while the system is going through a transition could have negative impacts. In particular, given that miners bid BTC in order to win the STX coinbase reward (and that BTC is what provides PoX payouts for Signers), it is highly likely that a 50% reduction in the STX coinbase reward would lead to a corresponding 50% reduction in PoX payouts, thereby dramatically decreasing incentives for signers while Nakamoto and sBTC are gaining adoption. Additionally, as discussed in the 7th Avenue Group report [1], by reducing the gross value of the block rewards and corresponding PoX payouts, it is likely that some miners and signers may choose to cease their work on the Stacks blockchain, reducing competition and economics further.
```
grammar fix

While ultimately halvings do need to happen, for those reasons, it would be highly preferable to not change the economic incentives until both Nakamoto and sBTC have been live for some time, as well as to have reductions in the STX block reward happen at the same time as reductions in the BTC block reward given their links.

> Therefore, this SIP proposes  altering the token emission schedule to preserve the existing incentive structure while ensuring no increase in the 2050 target supply, and incorporating these principles:

Reviewer: ```suggestion
Therefore, this SIP proposes altering the token emission schedule to preserve the existing incentive structure while ensuring no increase in the 2050 target supply, and incorporating these principles:
```
remove extra space

* All STX holders would see a reduced 2050 target STX supply by 0.19%, thereby making STX more scarce
* All STX holders would see a reduced tail inflation after the final halving
* Stacks miners and signers’ economic incentives would remain consistent with the existing incentives for the next 16 months

> * The new Stacks halving schedule would align with Bitcoin halvings starting in 2028 to strengthen the connection the Stacks L2 has to Bitcoin and also synchronize the economic adjustments of both assets, reducing changes in incentives for miners and signers at each halving, as further outlined on the Forum post [2]

Reviewer: ```suggestion
* The new Stacks halving schedule would align with Bitcoin halvings starting in 2028 to strengthen the connection the Stacks L2 has to Bitcoin and also synchronize the economic adjustments of both assets, reducing changes in incentives for miners and signers at each halving, as further outlined in the Forum post [2]
```
fix grammar

| Current                        | -                   | -                                   | 1000                 | -                 |
| 1st*                     | 945,000             | ~April 2026 (+1.33 yrs)            | 500 (50%)           | 1,607,907,038     |
| 2nd                     | 1,050,000           | ~April 2028 (2 yrs)                | 250 (50%)           | 1,652,242,188     |
| 3rd                      | 1,260,000           | ~April 2032 (4 yrs)                | 125 (50%)           | 1,696,547,013     |
| 4th                      | 1,470,000           | ~April 2036 (4 yrs)                | 100 (20%)           | 1,718,699,425     |
| -                          | -                   | ~Jan 2050 (13.833 yrs)             | 100 (0%)             | 1,779,628,415     |

With these changes, miner incentives and PoX yield remain unchanged for another 16 months, which we believe is sufficient time for the nascent Nakamoto signer set to develop into a set of stable, professional block signers, and for the sBTC project to attract sufficient initial liquidity.

> 

Reviewer: Missing 'Related Work' section: The SIP must include a 'Related Work' section that summarizes alternative solutions addressing similar problems and explains why they are not adequate.

Reviewer 2: @will-corcoran you can add links to the reports here, that should be sufficient, or explain why this section isn't relevant for this SIP

Reviewer 2: since i'm not aware of any precedence here @will-corcoran  - you may just want to include that we're not aware of any related work in this area. i've looked and cannot find anything

Author: @wileyj @jiga related work added

Reviewer 3: thank you @will-corcoran - very helpful to see the proposed STX emission schedule

- The Bitcoin address is the `base58check` of the hash of the Bitcoin script above.
- The Stacks address is the `c32check-encoded` message.

Stackers (pool and solo) vote by sending Stacks dust to the corresponding Stacks address from the account where their Stacks are locked.

Solo stackers only can also vote by sending a bitcoin dust transaction (6000 sats) to the corresponding bitcoin address.

#### For Non-Stackers:

Users with liquid STX can vote on proposals directly at [stx.eco](https://stx.eco) using the Ecosystem DAO. Liquid STX is the user’s balance, less any STX they have locked in the PoX stacking protocol, at the block height at which the voting started (preventing the same STX from being transferred between accounts and used to effectively double vote). This is referred to generally as "snapshot" voting.

For this SIP to pass, 80% of all liquid STX committed by voting must be in favor of the proposal.

We believe that these thresholds are sufficient to demonstrate interest from Stackers -- Stacks users who have a long-term interest in the Stacks blockchain's successful operation -- in performing this upgrade.

# Reference Implementation

> [To be added: Link to implementation PR]

Reviewer: reminder to add PR or Issue link

Reviewer 2: removed

Reviewer 2: @will-corcoran

Author: noted - thanks

> # Reference Implementation
> 
> [To be added: Link to implementation PR]

Reviewer 2: ```suggestion
```

> # References
> [1] Link to seventh ave sources. Annotate where it’s used above. Etc. https://forum.stacks.org/t/stacks-halving-schedule-reports-and-recommendations/15774
> [2] Link to the forum post
> https://forum.stacks.org/t/aligning-with-bitcoin-halving-and-incentives-after-nakamoto/17668

Reviewer 2: ```suggestion
# References
[1] Soslow J (2023) Review of Mining Emissions and Risks of the Halving. Available at https://stx.is/emissions-report-1 [Verified 5 November 2024]
[2] Laughlin, B (2024) Aligning with Bitcoin Halving and Incentives after Nakamoto. [Online forum post] forum.stacks.org https://forum.stacks.org/t/aligning-with-bitcoin-halving-and-incentives-after-nakamoto/17668 [Verified 5 November 2024]

```

# Preamble

**SIP Number:** 029

**Title:** Bootstrapping sBTC Liquidity and Nakamoto Signer Incentives

**Authors:**
- Alex Miller (alex@hiro.so)
- Andre Serrano (andre@bitcoinl2labs.com)
- Brittany Laughlin (brittany@lattice.vc)
- Jesse Wiley (jesse@stacks.org)
- Jude Nelson (jude@stacks.org)
- Philip De Smedt (philip@stackingdao.com)
- Tycho Onnasch (tycho@zestprotocol.com)
- Will Corcoran (will@stacks.org)

**Consideration:** Economic, Technical, Governance

> **Type:** Consensus (hard fork)

Reviewer 2: ```suggestion
**Type:** Consensus
**Layer:** Consensus (hard fork)
```

# Preamble

**SIP Number:** 029

**Title:** Bootstrapping sBTC Liquidity and Nakamoto Signer Incentives

**Authors:**
- Alex Miller (alex@hiro.so)
- Andre Serrano (andre@bitcoinl2labs.com)
- Brittany Laughlin (brittany@lattice.vc)
- Jesse Wiley (jesse@stacks.org)
- Jude Nelson (jude@stacks.org)
- Philip De Smedt (philip@stackingdao.com)
- Tycho Onnasch (tycho@zestprotocol.com)
- Will Corcoran (will@stacks.org)

**Consideration:** Economic, Technical, Governance

**Type:** Consensus

> **Layer:** Consensus (hard fork)

Reviewer 2: ```suggestion

**Layer:** Consensus (hard fork)
```

**Status:** Draft

**Created:** 2024-11-06

**License:** BSD 2-Clause

**Sign-off:**

**Discussions-To:**
- Stacks Forum Post: [Aligning with Bitcoin Halving and Incentives after Nakamoto](https://forum.stacks.org/t/aligning-with-bitcoin-halving-and-incentives-after-nakamoto/17668)

# Abstract

> The first Stacks halving is expected to take place at Stacks block height 210,384, which is set to occur during Reward Cycle 100 in December 2024, cutting the STX block reward from 1,000 STX to 500 STX. This SIP proposes a modification to the emissions schedule given that the network is going through two major launches (Nakamoto and sBTC) which rely on predictable economic incentives for chain security. The proposed schedule modification and associated STX emission rate would create time for Nakamoto and sBTC to launch and settle in, but, being mindful of supply, would still result in an overall reduced target 2050 STX supply (0.19% lower) and a reduced tail emission rate (20% lower).

Reviewer 2: ```suggestion
The first Stacks halving is expected to take place at Stacks block height 210,384, which is set to occur during Reward Cycle 100 in December 2024, cutting the STX block reward from 1,000 STX to 500 STX. This SIP proposes a modification to the emissions schedule given that the network is going through two major launches (Nakamoto and sBTC) which rely on predictable economic incentives. The proposed schedule modification and associated STX emission rate would create time for Nakamoto and sBTC to launch and settle in, but, being mindful of supply, would still result in an overall reduced target 2050 STX supply (0.19% lower) and a reduced tail emission rate (20% lower).
```

# License and Copyright

This SIP is made available under the terms of the BSD-2-Clause license,
available at https://opensource.org/licenses/BSD-2-Clause.  This SIP’s copyright
is held by the Stacks Open Internet Foundation.

# Introduction

> STX halvings were originally designed to happen every 4 years like Bitcoin. The first Stacks halving is expected to take place at Stacks block height 210,384, which is set to occur during Reward Cycle 100 in December 2024, cutting the STX block reward from 1,000 STX to 500 STX, thereby significantly altering the economic incentives of mining on the Stacks blockchain. At the same time, the Stacks network is going through two major upgrades/changes, both of which are highly dependent on economic incentives.

Reviewer 2: ```suggestion
STX halvings were originally designed to happen every 4 years similar to Bitcoin. The first Stacks halving is expected to take place at Stacks block height 210,384, which is set to occur during Reward Cycle 100 in December 2024, cutting the STX block reward from 1,000 STX to 500 STX, thereby significantly altering the economic incentives of mining on the Stacks blockchain. At the same time, the Stacks network is going through two major upgrades/changes, both of which are highly dependent on economic incentives.
```

First, the role of Stackers is changing. The role of Stackers in the Nakamoto blockchain (SIP-021) differs from their role in the original Stacks blockchain in that they must now collectively sign Stacks blocks. This is required to prevent Stacks forks from arising, and to secure the chain tip before it can be fully anchored to the Bitcoin chain. This also means that Stackers must run and maintain signing nodes with high availability in order to ensure that blocks always reach the requisite signature threshold. The PoX payouts (SIP-007) granted to Stackers by miners provide a positive monetary incentive for Stackers to carry out this task.

> Second, the release of the Nakamoto blockchain unblocks sBTC (SIP-028), a wrapped Bitcoin asset on the Stacks chain which benefits from Nakamoto’s newfound resistance to forks. Because inducing a chain reorganization (a reorg) in Nakamoto is at least as hard as doing so in the Bitcoin chain, applications can rely on Nakamoto to ensure that each sBTC token is backed 1:1 by a BTC token – there is no feasible way to double-spend sBTC through a Stacks chain reorg.

Reviewer 2: ```suggestion
Second, the release of the Nakamoto blockchain unblocks sBTC (SIP-028), a wrapped Bitcoin asset on the Stacks blockchain which benefits from Nakamoto’s newfound resistance to forks. Because inducing a chain reorganization (a reorg) in Nakamoto is at least as hard as doing so in the Bitcoin blockchain, applications can rely on Nakamoto to ensure that each sBTC token is backed 1:1 by a BTC token – there is no feasible way to double-spend sBTC through a Stacks blockchain reorg.
```

> Because of how important both of these upgrades are to the future of the Stacks blockchain and the dependency of all blockchains on economic incentives for security, changing the incentives while the system is going through a transition could have negative impacts. In particular, given that miners bid BTC in order to win the STX coinbase reward (and that BTC is what provides PoX payouts for Signers), it is highly likely that a 50% reduction in the STX coinbase reward would lead to a corresponding 50% reduction in PoX payouts, thereby dramatically decreasing incentives for signers while Nakamoto and sBTC are gaining adoption. Additionally, as discussed in the 7th Avenue Group report [1], by reducing the gross value of the block rewards and corresponding PoX payouts, it is likely that some miners and signers may choose to cease their work on the Stacks blockchain, reducing competition and economics further.

Reviewer 2: ```suggestion
Because of how important both of these upgrades are to the future of the Stacks blockchain and the dependency of all blockchains on economic incentives for security, changing the incentives while the system is going through a transition could have negative impacts. In particular, given that miners bid BTC in order to win the STX coinbase reward (and that BTC is what provides PoX payouts for Signers), it is highly likely that a 50% reduction in the STX coinbase reward could lead to a corresponding 50% reduction in PoX payouts, thereby dramatically decreasing incentives for signers while Nakamoto and sBTC are gaining adoption. Additionally, as discussed in the 7th Avenue Group report [1], by reducing the gross value of the block rewards and corresponding PoX payouts, it is likely that some miners and signers may choose to cease their work on the Stacks blockchain, reducing competition and economics further.
```

# Preamble

**SIP Number:** 029

> **Title:** Bootstrapping sBTC Liquidity and Nakamoto Signer Incentives

Reviewer 2: hmm, i'm not sure the first part of the title suggestion is entirely accurate. 
the goal of the SIP as i read it is to ensure current incentives remain during sbtc bootstrap, and give nakamoto more time to settle in before changes are effected. 

the schedule change to align with bitcoin is secondary, but not a requirement of this sip (as i read it).

what do you think about a change like this:
```
**Title:** Bootstrapping sBTC Liquidity and Incentive Preservation
```

> **Title:** Bootstrapping sBTC Liquidity and Nakamoto Signer Incentives

Author: I like `Bootstrapping sBTC Liquidity and Incentive Preservation` - thoughts @friedger

> **Title:** Bootstrapping sBTC Liquidity and Nakamoto Signer Incentives

Reviewer 4: The specification does not say anything about sBTC. I don't have strong opinion here.

> **Title:** Bootstrapping sBTC Liquidity and Nakamoto Signer Incentives

Reviewer 4: I just took the Title of this PR as a suggested title for the SIP

> **Title:** Bootstrapping sBTC Liquidity and Nakamoto Signer Incentives

Author: I spoke with some of the other co-authors and landed on a more appropriate titled "Preserving Economic Incentives During Stacks Network Upgrades".

The SIP and the PR have been updated to reflect this change.

While ultimately halvings do need to happen, for those reasons, it would be highly preferable to not change the economic incentives until both Nakamoto and sBTC have been live for some time, as well as to have reductions in the STX block reward happen at the same time as reductions in the BTC block reward given their links.

Therefore, this SIP proposes altering the token emission schedule to preserve the existing incentive structure while ensuring no increase in the 2050 target supply, and incorporating these principles:

* All STX holders would see a reduced 2050 target STX supply by 0.19%, thereby making STX more scarce
* All STX holders would see a reduced tail inflation after the final halving
* Stacks miners and signers’ economic incentives would remain consistent with the existing incentives for the next 16 months
* The new Stacks halving schedule would align with Bitcoin halvings starting in 2028 to strengthen the connection the Stacks L2 has to Bitcoin and also synchronize the economic adjustments of both assets, reducing changes in incentives for miners and signers at each halving, as further outlined in the Forum post [2]

> # Specification

Reviewer 4: The specification and the SIP in general is missing details about the hard fork. 
I guess that we will have Stacks 3.1

Reviewer 2: @will-corcoran

Author: @friedger @wileyj thanks for the note.  I will add the following text to the beginning of the Specification section:

> Applying these upgrades to the Stacks blockchain requires a consensus-breaking network upgrade, in this case, a hard fork. Like other such changes, this will require a new Stacks epoch. In this SIP, we will refer to this new epoch as Stacks 3.1.

| 2nd                     | 1,050,000           | ~April 2028 (2 yrs)                | 250 (50%)           | 1,652,242,188     |
| 3rd                      | 1,260,000           | ~April 2032 (4 yrs)                | 125 (50%)           | 1,696,547,013     |
| 4th                      | 1,470,000           | ~April 2036 (4 yrs)                | 100 (20%)           | 1,718,699,425     |
| -                          | -                   | ~Jan 2050 (13.833 yrs)             | 100 (0%)             | 1,779,628,415     |

With these changes, miner incentives and PoX yield remain unchanged for another 16 months, which we believe is sufficient time for the nascent Nakamoto signer set to develop into a set of stable, professional block signers, and for the sBTC project to attract sufficient initial liquidity.

# Related Work

While many blockchain protocols implement token emission schedules with periodic reductions in block rewards, we are not aware of any precedent for modifying an emission schedule specifically to maintain economic incentives during major protocol upgrades.

# Backwards Compatibility

This change requires a hard fork of the Stacks blockchain

# Activation

> 

Reviewer 4: It is not clear how this sip activates. I like the formula of SIP 21

> There are different rules for activating this SIP.

Stacks holders need to vote, miners need to mine.

Author: updated to address your comment @friedger

## Voting Timeline

Voting will begin at bitcoin block height 870,750, which occurs ~ Sunday, November 17th, 2024.
Voting will conclude at bitcoin block height 872,750, which occurs ~ Sunday, December 1st, 2024.

## Activation

The SIP-029 STX emission schedule is designed to activate on Stacks 3.0 as defined in [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md). Therefore, this SIP is only meaningful when SIP-021 activates.

### Process of Activation

Users can vote to approve this SIP with either their locked/stacked STX or with unlocked/liquid STX, or both. The SIP voting page can be found at [stx.eco](https://stx.eco). The criteria for the stacker and non-stacker voting is as follows.

#### For Stackers:

In order for this SIP to activate, the following criteria must be met by the set of Stacked STX:

> - At least 80 million Stacked STX must vote, with at least 80% (64 million) voting "yes".

Reviewer 4: What if 100 million STX voted "no"?

Reviewer 2: then the sip isn't approved. 
here, the floor requirement is that 80 million stacked stx *must* vote. if there is more, even better. of those that vote (assuming the threshold of 80 million is reached), at least 80% must vote yes. 

if 100 million vote no in your scenario, then it would take another 400 million voting yes (at a minimum)

> - At least 80 million Stacked STX must vote, with at least 80% (64 million) voting "yes".

Reviewer 4: ```suggestion
- At least 80 million Stacked STX must vote, with at least 80% of all stacked STX committed by voting must be in favor of the proposal (vote "yes").
```

| - | - | - | - | - | - |
| yes      | `11111111111mdWK2VXcrA1e7in77Ux` | `SP00000000001WPAWSDEDMQ0B9J76GZNR3T` | `yes-sip-29` | `000000000000000000007965732d7369702d3239` | `OP_DUP` `OP_HASH160` `000000000000000000007965732d7369702d3239` `OP_EQUALVERIFY` `OP_CHECKSIG` |
| no       | `111111111111ACW5wa4RwyeKgtEJz3` | `SP000000000006WVSDEDMQ0B9J76NCZPNZ`  | `no-sip-29`  | `00000000000000000000006e6f2d7369702d3239` | `OP_DUP` `OP_HASH160` `00000000000000000000006e6f2d7369702d3239` `OP_EQUALVERIFY` `OP_CHECKSIG` |

The addresses have been generated as follows:

- Encode `<message>` in ASCII, with 0-padding.
- Use the resulting `<encoding>` in the Bitcoin script`OP_DUP` `OP_HASH160` `<encoding>` `OP_EQUALVERIFY` `OP_CHECKSIG`.
- The Bitcoin address is the `base58check` of the hash of the Bitcoin script above.
- The Stacks address is the `c32check-encoded` message.

Stackers (pool and solo) vote by sending Stacks dust to the corresponding Stacks address from the account where their Stacks are locked.

Solo stackers only can also vote by sending a bitcoin dust transaction (6000 sats) to the corresponding bitcoin address.

#### For Non-Stackers:

> Users with liquid STX can vote on proposals directly at [stx.eco](https://stx.eco) using the Ecosystem DAO. Liquid STX is the user’s balance, less any STX they have locked in the PoX stacking protocol, at the block height at which the voting started (preventing the same STX from being transferred between accounts and used to effectively double vote). This is referred to generally as "snapshot" voting.

Reviewer 4: @will-corcoran Has the proposal been deployed?

Author: @friedger can you clarify what you mean by 'deployed'? 

cc: @Hero-Gamer

Reviewer 2: > @friedger can you clarify what you mean by 'deployed'?
> 
> cc: @Hero-Gamer

the question is if stx.eco has a section to vote on this SIP (it does not)

Reviewer 5: Will this vote be simplified to just have non-stackers also send a dust amount to the STX addresses to vote?

Author: Hi @obycode - the intention is to have this vote run just like the SIP-028 vote (cc: @radicleart @Hero-Gamer )

@friedger - Mike is in the process of deplying the proposal.  The plan is to have all CAB votes completed by EOD Wed Nov 13 and then Mike will get the platform set up to host this vote.  The vote would then open on Sunday Nov 17.

Reviewer 5: > the intention is to have this vote run just like the SIP-028 vote

Right, didn't that remove the DAO voting and just use the send to address voting for everyone (though it looks like the SIP text was not updated accordingly).

Reviewer 6: Yes - your right @obycode the mention of Ecosystem Dao is confusing. It should say something like

"Users voting power is the sum of their locked and unlocked balance, calculated from the stacks transaction, at the height when voting began. User's can vote by visiting the [stx.eco](https://stx.eco) platform."

Author: thanks @radicleart I will updated the SIP to reflect your suggested revision later this afternoon.  @obycode are you good with this revised wording?

Reviewer 6: Maybe simpler to merge the "For Stackers" and "Non Stackers" into one section to reflect that the same tx counts both their locked and unlocked balances.

Reviewer 5: Agreed. Here's my suggestion for modifications (deleting the "For Non-Stackers:" section):

```md
### Process of Activation

 Users can vote to approve this SIP with either their locked/stacked STX or with unlocked/liquid STX, or both. The SIP voting page can be found at [stx.eco](https://stx.eco). The criteria for the stacker and non-stacker voting is as follows.

 #### To Vote:

 In order for this SIP to activate, the following criteria must be met:

 - At least 80 million stacked STX must vote, with at least 80% of all stacked STX committed by voting must be in favor of the proposal (vote "yes").
- At least 80% of all liquid STX committed by voting must be in favor of the proposal (vote "yes").

 The voting addresses will be:

 | Vote | Bitcoin Address | Stacks Address | Msg | ASCII-encoded msg | Bitcoin script |
 | - | - | - | - | - | - |
 | yes      | `11111111111mdWK2VXcrA1e7in77Ux` | `SP00000000001WPAWSDEDMQ0B9J76GZNR3T` | `yes-sip-29` | `000000000000000000007965732d7369702d3239` | `OP_DUP` `OP_HASH160` `000000000000000000007965732d7369702d3239` `OP_EQUALVERIFY` `OP_CHECKSIG` |

 | no       | `111111111111ACW5wa4RwyeKgtEJz3` | `SP000000000006WVSDEDMQ0B9J76NCZPNZ`  | `no-sip-29`  | `00000000000000000000006e6f2d7369702d3239` | `OP_DUP` `OP_HASH160` `00000000000000000000006e6f2d7369702d3239` `OP_EQUALVERIFY` `OP_CHECKSIG` |

 The addresses have been generated as follows:

 - Encode `<message>` in ASCII, with 0-padding.
 - Use the resulting `<encoding>` in the Bitcoin script`OP_DUP` `OP_HASH160` `<encoding>` `OP_EQUALVERIFY` `OP_CHECKSIG`.
 - The Bitcoin address is the `base58check` of the hash of the Bitcoin script above.
 - The Stacks address is the `c32check-encoded` message.

All STX holders vote by sending Stacks dust to the corresponding Stacks address from the account where their Stacks are held (stacked or liquid). Voting power is determined by a snapshot of the amount of STX (stacked and un-stacked) at the block height at which the voting started (preventing the same STX from being transferred between accounts and used to effectively double vote).

 Solo stackers only can also vote by sending a bitcoin dust transaction (6000 sats) to the corresponding bitcoin address.
```

Author: awesome.  thanks, @obycode and @radicleart

Reviewer 5: Thanks!

For this SIP to pass, 80% of all liquid STX committed by voting must be in favor of the proposal.

We believe that these thresholds are sufficient to demonstrate interest from Stackers -- Stacks users who have a long-term interest in the Stacks blockchain's successful operation -- in performing this upgrade.

# Reference Implementation

[To be added: Link to implementation PR]

# References

[1] Soslow J (2023) Review of Mining Emissions and Risks of the Halving. Available at https://stx.is/emissions-report-1 [Verified 5 November 2024]

> [2] Laughlin, B (2024) Aligning with Bitcoin Halving and Incentives after Nakamoto. [Online forum post] forum.stacks.org https://forum.stacks.org/t/aligning-with-bitcoin-halving-and-incentives-after-nakamoto/17668 [Verified 5 November 2024]

Reviewer 4: ```suggestion

[2] Laughlin, B (2024) Aligning with Bitcoin Halving and Incentives after Nakamoto. [Online forum post] forum.stacks.org https://forum.stacks.org/t/aligning-with-bitcoin-halving-and-incentives-after-nakamoto/17668 [Verified 5 November 2024]
```

Reviewer 4: Extra line needed for line break

Reviewer 2: good catch. committing

Therefore, this SIP proposes altering the token emission schedule to preserve the existing incentive structure while ensuring no increase in the 2050 target supply, and incorporating these principles:

* All STX holders would see a reduced 2050 target STX supply by 0.77%, thereby making STX more scarce
* All STX holders would see a reduced tail inflation (from 125 STX to 62.5 STX) after the final halving
* Stacks miners and signers’ economic incentives would remain consistent with the existing incentives for the next 16 months
* The new Stacks halving schedule would align with Bitcoin halvings starting in 2028 to strengthen the connection the Stacks L2 has to Bitcoin and also synchronize the economic adjustments of both assets, reducing changes in incentives for miners and signers at each halving, as further outlined in the Forum post [3]

# Specification

Applying these upgrades to the Stacks blockchain requires a consensus-breaking network upgrade, in this case, a hard fork. Like other such changes, this will require a new Stacks epoch. In this SIP, we will refer to this new epoch as Stacks 3.1.

> The _current_ STX emission schedule is presented as follows.  Note that the **first STX halving is in December 2024**.  The tail emission after the final halving in 2050 would be 125 STX per block.

Reviewer 5: ```suggestion
The _current_ STX emission schedule is presented as follows.  Note that the **first STX halving is in December 2024**.  The tail emission after the final halving in 2032 would be 125 STX per block.
```

| 4th                | 1,470,000      | Apr 2036 (4y)       | 1,758,602,847    | 62.5 (50%)   | 0.37%          |
| -                  | 2,197,560      | Jan 2050 (13.83y)   | 1,804,075,347    | 62.5 (0%)   | 0.18%          |

With these changes, miner incentives and PoX yield remain unchanged for another 16 months, which we believe is sufficient time for the nascent Nakamoto signer set to develop into a set of stable, professional block signers, and for the sBTC project to attract sufficient initial liquidity.

The model for both the current and proposed emissions schedules can be found in the SIP-029 Stacks Emission Model[4].

# Related Work

While many blockchain protocols implement token emission schedules with periodic reductions in block rewards, we are not aware of any precedent for modifying an emission schedule specifically to maintain economic incentives during major protocol upgrades.

# Backwards Compatibility

This change requires a hard fork of the Stacks blockchain

# Activation

## Voting Timeline

Voting will begin at bitcoin block height 870,750, which occurs ~ Sunday, November 17th, 2024.

> Voting will conclude at bitcoin block height 872,750, which occurs ~ Sunday, December 1st, 2024.

Reviewer 7: When does this new schedule activate?  Can we get a target activation height?

I'd propose 876,000, which is about 22 days after voting closes.

# Preamble

**SIP Number:** 029

**Title:** Bootstrapping sBTC Liquidity and Incentive Preservation

**Authors:**
- Alex Miller (alex@hiro.so)
- Andre Serrano (andre@bitcoinl2labs.com)
- Brittany Laughlin (brittany@lattice.vc)
- Jesse Wiley (jesse@stacks.org)
- Jude Nelson (jude@stacks.org)
- Philip De Smedt (philip@stackingdao.com)
- Tycho Onnasch (tycho@zestprotocol.com)
- Will Corcoran (will@stacks.org)

**Consideration:** Economics, Technical, Governance

**Type:** Consensus (hard fork)

**Status:** Draft

**Created:** 2024-11-06

**License:** BSD 2-Clause

**Sign-off:**

**Discussions-To:**
- Stacks Forum Post: [Aligning with Bitcoin Halving and Incentives after Nakamoto](https://forum.stacks.org/t/aligning-with-bitcoin-halving-and-incentives-after-nakamoto/17668)

# Abstract

> The first Stacks halving is expected to take place at Stacks block height 210,384, which is set to occur during Reward Cycle 100 in December 2024, cutting the STX block reward from 1,000 STX to 500 STX. This SIP proposes a modification to the emissions schedule given that the network is going through two major launches (Nakamoto and sBTC) which rely on predictable economic incentives. The proposed schedule modification and associated STX emission rate would create time for Nakamoto and sBTC to launch and settle in, but, being mindful of supply, would still result in an overall reduced target 2050 STX supply (0.19% lower) and a reduced tail emission rate (20% lower).

Reviewer 5: > at Stacks block height 210,384
This is not correct. It is set for 210,384 Bitcoin blocks after the Stacks 2.0 starting height, 666,050, which is Bitcoin height 876,434.

Author: Noted. I will correct this now.

# License and Copyright

This SIP is made available under the terms of the BSD-2-Clause license,
available at https://opensource.org/licenses/BSD-2-Clause.  This SIP’s copyright
is held by the Stacks Open Internet Foundation.

# Introduction

> STX halvings were originally designed to happen every 4 years, similar to Bitcoin. The first Stacks halving is expected to take place at Stacks block height 210,384, which is set to occur during Reward Cycle 100 in December 2024, cutting the STX block reward from 1,000 STX to 500 STX, thereby significantly altering the economic incentives of mining on the Stacks blockchain. At the same time, the Stacks network is going through two major upgrades/changes, both of which are highly dependent on economic incentives.

Reviewer 4: ```suggestion
STX halvings were originally designed to happen every 4 years, similar to Bitcoin. The first Stacks halving is expected to take place at Stacks tenure height 210,384, which is set to occur during Reward Cycle 100 in December 2024, cutting the STX block reward from 1,000 STX to 500 STX, thereby significantly altering the economic incentives of mining on the Stacks blockchain. At the same time, the Stacks network is going through two major upgrades/changes, both of which are highly dependent on economic incentives.
```

Reviewer 7: This text is wrong.  The first Stacks halvening was targeted to take place at Bitcoin height 876,430.  Stacks block heights (or tenure heights) have nothing to do with it.

Reviewer 2: good catch, i missed that on my readthrough. @will-corcoran jude is correct here and the text should reference the bitcoin block, not the stacks block (especially post 3.0, we can't be certain which stacks block it will be).

Reviewer 2: unless my calculation is off, it looks like btc block 876434

https://github.com/stacks-network/stacks-core/blob/master/stackslib/src/chainstate/stacks/db/blocks.rs#L3638

which should arrive in late december based on current block times.

Reviewer 5: Yup, I arrived at the same conclusion above before seeing your comment here.

Author: Noted.  I will correct this now.  Thanks @jcnelson @wileyj @obycode

> STX halvings were originally designed to happen every 4 years, similar to Bitcoin. The first Stacks halving is expected to take place at Stacks block height 210,384, which is set to occur during Reward Cycle 100 in December 2024, cutting the STX block reward from 1,000 STX to 500 STX, thereby significantly altering the economic incentives of mining on the Stacks blockchain. At the same time, the Stacks network is going through two major upgrades/changes, both of which are highly dependent on economic incentives.

Reviewer 4: ```suggestion
STX halvings were originally designed to happen every 4 years, similar to Bitcoin. The first Stacks halving is expected to take place at Bitcoin block 876,434, which is set to occur during Reward Cycle 100 in December 2024, cutting the STX block reward from 1,000 STX to 500 STX, thereby significantly altering the economic incentives of mining on the Stacks blockchain. At the same time, the Stacks network is going through two major upgrades/changes, both of which are highly dependent on economic incentives.
```

Author: noted.  thanks

Because of how important both of these upgrades are to the future of the Stacks blockchain and the dependency of all blockchains on economic incentives for security, changing the incentives while the system is going through a transition could have negative impacts. In particular, given that miners bid BTC in order to win the STX coinbase reward (and that BTC is what provides PoX payouts for Signers), it is highly likely that a 50% reduction in the STX coinbase reward would lead to a corresponding 50% reduction in PoX payouts, thereby dramatically decreasing incentives for signers while Nakamoto and sBTC are gaining adoption. Additionally, as discussed in the 7th Avenue Group report [1][2], by reducing the gross value of the block rewards and corresponding PoX payouts, it is likely that some miners and signers may choose to cease their work on the Stacks blockchain, reducing competition and economics further.

> Additionally, the increased block production rate under Nakamoto (from 122 to 144 blocks per day) means that without modification, the 2050 supply would exceed the accepted supply cap of 1,818,000,000 STX.

Reviewer 5: This is not correct. The amount of STX created was only ever dependent on the number of Bitcoin blocks, not the number of Stacks blocks. I think this paragraph can just be deleted.

Therefore, this SIP proposes altering the token emission schedule to preserve the existing incentive structure while ensuring no increase in the 2050 target supply, and incorporating these principles:

* All STX holders would see a reduced 2050 target STX supply by 0.77%, thereby making STX more scarce
* All STX holders would see a reduced tail inflation (from 125 STX to 62.5 STX) after the final halving
* Stacks miners and signers’ economic incentives would remain consistent with the existing incentives for the next 16 months
* The new Stacks halving schedule would align with Bitcoin halvings starting in 2028 to strengthen the connection the Stacks L2 has to Bitcoin and also synchronize the economic adjustments of both assets, reducing changes in incentives for miners and signers at each halving, as further outlined in the Forum post [3]

# Specification

Applying these upgrades to the Stacks blockchain requires a consensus-breaking network upgrade, in this case, a hard fork. Like other such changes, this will require a new Stacks epoch. In this SIP, we will refer to this new epoch as Stacks 3.1.

> The _current_ STX emission schedule is presented as follows.  Note that the **first STX halving is in December 2024**.  The tail emission after the final halving in 2050 would be 125 STX per block, and due to the increased block production rate under Nakamoto (144 blocks per day vs the previous 122), the total supply at that time is projected to be 1,829,119,597 STX, which exceeds the accepted supply cap of 1,818,000,000 STX (0.61% higher).

Reviewer 5: Again, I think just remove the stuff about blocks per day since that does not affect the supply. I am not sure why the original estimate was off, but that is not why.

```suggestion
The _current_ STX emission schedule is presented as follows.  Note that the **first STX halving is in December 2024**.  The tail emission after the final halving in 2050 would be 125 STX per block.
```

Reviewer 8: The original estimate was off because it presumed launching Stacks 2.0 with the bitcoin halving in 2020. Instead it was launched in early January 2021 and the schedule wasn't adjusted to to sync up exactly with bitcoin halvings. Halvings are  merely 4, 8 and 12 years after the launch of Stacks 2.0 without the proposed adjustment in SIP-029. @will-corcoran In case you want to add something about the real reason.

|--------------------------------|-------------------------------------|----------------------|-------------------|
| Current                         | -                                   | 1000                 | -                  |
| 1st                             | ~Dec 2024 (4 yrs)                  | 500 (50%)           | 1,512,993,315      |
| 2nd                             | ~Dec 2028 (4 yrs)                  | 250 (50%)           | 1,645,084,888      |
| 3rd                             | ~Dec 2032 (4 yrs)                  | 125 (50%)           | 1,689,389,675      |
| -                               | ~Jan 2050 (17.08 yrs)             | 125 (0%)             | 1,783,063,600      |

> The _proposed_ STX emission schedule is presented as follows.  In particular, this SIP proposes preserving the 1000 STX coinbase until April 2026.  After this, there would be a reduction to 500 STX after two years, and another reduction to 125 STX after two more years.  The tail emission after the final halving in 2050 would be 100 STX, and the total supply at that time is projected to be 1,783,063,600 STX (about 0.19% lower).

Reviewer 5: Also, let's avoid using the word "halving" when the amount is reducing by something other than half.

> The _proposed_ STX emission schedule is presented as follows.  In particular, this SIP proposes preserving the 1000 STX coinbase until April 2026.  After this, there would be a reduction to 500 STX after two years, and another reduction to 125 STX after two more years.  The tail emission after the final halving in 2050 would be 100 STX, and the total supply at that time is projected to be 1,783,063,600 STX (about 0.19% lower).

Author: noted

> The _proposed_ STX emission schedule is presented as follows.  In particular, this SIP proposes preserving the 1000 STX coinbase until April 2026.  After this, there would be a reduction to 500 STX after two years, and another reduction to 125 STX after two more years.  The tail emission after the final reduction in 2036 would be 100 STX, and the total supply at that time is projected to be 1,783,063,600 STX (about 0.19% lower).

Reviewer 5: ```suggestion
The _proposed_ STX emission schedule is presented as follows.  In particular, this SIP proposes preserving the 1000 STX coinbase until April 2026.  After this, there would be a reduction to 500 STX after two years, and another reduction to 125 STX after two more years.  The tail emission after the final reduction in 2036 would be 100 STX, arriving at the total supply in January 2050 of 1,783,063,600 STX (about 0.19% lower than the current schedule).
```

I think this sentence needs this adjustment, then also fixing the numbers after confirming the correct values.

Reviewer 5: Replaced this comment with an updated suggestion.

> The _proposed_ STX emission schedule is presented as follows.  In particular, this SIP proposes preserving the 1000 STX coinbase until April 2026.  After this, there would be a halving to 500 STX after two years, and another halving to 125 STX after two more years.  The tail emission after the final halving in 2036 would be 62.5 STX, and the total supply at that time is projected to be 1,758,602,847 STX (about 0.27% lower).

Reviewer 5: ```suggestion
The _proposed_ STX emission schedule is presented as follows.  In particular, this SIP proposes preserving the 1000 STX coinbase until April 2026.  After this, there would be a halving to 500 STX after two years, aligning the remaining Stacks halving block heights with Bitcoin's halving schedule. Four years later, emissions halve again to 125 STX, and the tail emission after the final halving in 2036 would be 62.5 STX. This brings the total supply in 2050 to 1,804,075,347, or about 0.77% lower than the original 1,818,000,000 estimate.
```

| Coinbase Reward Phase | Bitcoin Height | Approx Date           | STX Supply at Block | STX Reward | Annual Inflation |
|--------------------|----------------|---------------------|-------------------|------------|-----------------|
| Current            | 870,100        | -                   | 1,552,452,847    | 1000       | -               |
| 1st                | 945,000        | Apr 2024 (+1.33y)   | 1,627,352,847    | 500 (50%)  | 3.23%          |
| 2nd                | 1,050,000      | Apr 2028 (4y)       | 1,679,852,847    | 250 (50%)  | 1.57%          |
| 3rd                | 1,260,000      | Apr 2032 (4y)       | 1,732,352,847    | 125 (50%)  | 0.76%          |
| 4th                | 1,470,000      | Apr 2036 (4y)       | 1,758,602,847    | 62.5 (50%)   | 0.37%          |

> | -                  | 2,197,560      | Jan 2050 (13.83y)   | 1,813,169,847    | 62.5 (0%)   | 0.18%          |

Reviewer 5: ```suggestion
| -                  | 2,197,560      | Jan 2050 (13.83y)   | 1,804,075,347    | 62.5 (0%)   | 0.18%          |
```

For this SIP to pass, 80% of all liquid STX committed by voting must be in favor of the proposal.

We believe that these thresholds are sufficient to demonstrate interest from Stackers -- Stacks users who have a long-term interest in the Stacks blockchain's successful operation -- in performing this upgrade.

#### For Miners

There is only one criterion for miners to activate this SIP: they must mine the Stacks blockchain up to and past the end of the voting period (bitcoin block height 872,750). In all reward cycles between cycle 97 & 98 and the end of the voting period, PoX must activate.

# Reference Implementation

[To be added: Link to implementation PR]

# References

[1] Soslow, J (2023) Review of Mining Emissions and Risks of the Halving. Available at https://stx.is/emissions-report-1 [Verified 5 November 2024]

[2] Laughlin, B (2024) Aligning with Bitcoin Halving and Incentives after Nakamoto. [Online forum post] forum.stacks.org https://forum.stacks.org/t/aligning-with-bitcoin-halving-and-incentives-after-nakamoto/17668 [Verified 5 November 2024]

> [3] Müffke, F & Corcoran, W (2024) SIP-029 Stacks Emission Model. [Online spreadsheet] docs.google.com https://docs.google.com/spreadsheets/d/1ZRQgQV99kWvcSjkmZWKgldflcB2ytaN6sjo2RiHcjnk/edit?gid=0#gid=0 [Verified 13 November 2024]

Reviewer: These documents should be added as supplemental material within same sip-029 folder, as specified in sip-000 here https://github.com/stacksgov/sips/blob/main/sips/sip-000/sip-000-stacks-improvement-proposal-process.md#supplemental-materials
